examples/goto/goto3.pvl

```diff
@@ -0,0 +1,31 @@
+// -*- tab-width:2 ; indent-tabs-mode:nil -*-
+//:: cases Goto3
+//:: tools silicon
+//:: verdict Pass
+
+class C {
+   int m() {
+     int r = 3;
+     goto lbl;
+     r = 4;
+     label lbl;
+     assert r == 3;
+   }
+}
+
+class D {
+  int m() {
+    int r = 10;
+
+    loop_invariant r == 10;
+    while (true) {
+      r = 20;
+      goto lbl2;
+    }
+    assert r == 10;
```

Comment:
Yes, you can (and it will verify). Since the while condition is always true it will never break, so goto lbl2 is guaranteed to be encountered. Therefore an `assert false` after line 25 will not trigger since it is impossible to get there. I guess this makes the `assert r == 10` less useful though.